src/spv/bitcoin/BRPeerManager.cpp

```diff
@@ -1431,27 +1447,30 @@ static void _peerRelayedBlock(void *info, BRMerkleBlock *block)
                 if (b) timestamp = timestamp/2 + b->timestamp/2;
                 if (count > 0) BRWalletUpdateTransactions(manager->wallet, txHashes, count, height, timestamp, b->blockHash);
             }
-        
+
+            manager->lock.lock();
             manager->lastBlock = block;
             
             if (block->height == manager->estimatedHeight) { // chain download is complete
                 saveCount = (block->height % BLOCK_DIFFICULTY_INTERVAL) + BLOCK_DIFFICULTY_INTERVAL + 1;
                 _BRPeerManagerLoadMempools(manager);
             }
+            manager->lock.unlock();
         }
     }
 
     // Has last block changed?
-    if (!UInt256Eq(lastBlockHash, manager->lastBlock->blockHash))
+    if (!UInt256Eq(lastBlockHash, lastBlockHash))
```

Comment:
This will always be false. The test here is to see whether `manager->lastBlock` was updated in the preceding conditionals.

Reply:
Will try another approach, this one is a bit aggressive.